### ISSUE TITLE
OS X: Cannot read fonts in ~/Library/Fonts

### ISSUE THREAD

**Issue by deepakjois** [deepakjois/luafontmanager#7]
Cannot read the fonts in the user’s home directory.

Need to check this for Linux as well, if applicable.

**Comment by deepakjois** [comment]
* lltxplatform uses [CTFontManagerCopyAvailableFontURLs()](https://developer.apple.com/reference/coretext/1499478-ctfontmanagercopyavailablefontur). As seen [here](https://github.com/deepakjois/luafontmanager/blob/291c65b442a96ec795b2d748c9556deb8fceeca1/src/luafontmanager/fonts_coretext.c#L34).
* font-manager uses [CTFontCollectionCreateFromAvailableFonts(_:)](https://developer.apple.com/reference/coretext/1509907-ctfontcollectioncreatefromavaila). As seen [here](https://developer.apple.com/reference/coretext/1509907-ctfontcollectioncreatefromavaila).

Not sure if `CTFontCollectionCreateFromAvailableFonts` will get us what we need.

**Comment by deepakjois** [comment]
It does not look like `CTFontCollectionCreateFromAvailableFonts` will get us what we need. Filed devongovett/font-manager#19 to find out more about whether there is an API that can read fonts from the user’s home directory.

**Comment by michal-h21** [comment]
I dont know if the following idea is really usefull, but it seems that fc-list utility which comes with TeX Live (I dont know about Miktex, but I quess it should be similar) can list all fonts installed both in system and in TeX tree. It can also be used to get family name, style and other information about particular font files. 

It works at least on Linux and Windows, I cannot test it on OS X. Maybe it could be possible to use it as a library even without TeX distribution?

**Comment by deepakjois** [comment]
Yes, that is an option. `fc-list` is actually part of [fontconfig](https://www.freedesktop.org/wiki/Software/fontconfig/). 

Instead of depending on a commandline tool like `fc-list` it might be better to just use the fontconfig API.  In fact, on Linux, we use fontconfig to get this information already because fontconfig is pretty much guaranteed to be pre-installed on any decent Linux desktop. 

fontconfig can be used as a library on OS X as well, just like Harfbuzz.  I was just a bit reluctant to add an additional dependency on OS X for fontconfig. It will mean that before installing this package from LuaRocks, the user will need to have fontconfig headers and library installed in the right place.

I am not sure about Windows yet. So maybe there, as a fallback, we can use tools like `fc-list` or the native Windows directwrite backend to retrieve fonts.

**Comment by deepakjois** [comment]
Pushed a [change on another branch](https://github.com/ufyTeX/luafontmanager/tree/fcd90125841af5b355d089edad6cacf99b024513), which makes OS X use fontconfig as well. It works well and detects all the fonts, even the ones in the home folder.

**Comment by michal-h21** [comment]
`fontmanager` could then return some additional field, if `fontconfig` could be used on all platforms. I can think at least of `family`, `style` and `fontformat`. It would then enable to select font family and then use particular font files for standard TeX style and shape commands.
